azurerm/resource_arm_batch_account.go

```diff
@@ -0,0 +1,231 @@
+		parameters.AccountCreateProperties.AutoStorage = &batch.AutoStorageBaseProperties{
+			StorageAccountID: &storageAccountId,
+		}
+	}
+
+	future, err := client.Create(ctx, resourceGroupName, name, parameters)
+	if err != nil {
+		return fmt.Errorf("Error creating Batch account %q (Resource Group %q): %+v", name, resourceGroupName, err)
+	}
+
+	err = future.WaitForCompletionRef(ctx, client.Client)
+	if err != nil {
+		return fmt.Errorf("Error waiting for creation of Batch account %q (Resource Group %q): %+v", name, resourceGroupName, err)
+	}
+
+	read, err := client.Get(ctx, resourceGroupName, name)
+	if err != nil {
+		return fmt.Errorf("Error retrieving Batch account %q (Resource Group %q): %+v", name, resourceGroupName, err)
+	}
+
+	if read.ID == nil {
+		return fmt.Errorf("Cannot read Batch account %q (resource group %q) ID", name, resourceGroupName)
+	}
+
+	d.SetId(*read.ID)
+
+	return resourceArmBatchAccountRead(d, meta)
+}
+
+func resourceArmBatchAccountRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).batchAccountClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	name := id.Path["batchAccounts"]
+	resourceGroupName := id.ResourceGroup
+
+	resp, err := client.Get(ctx, resourceGroupName, name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			d.SetId("")
+			log.Printf("[DEBUG] Batch Account %q was not found in Resource Group %q - removing from state!", name, resourceGroupName)
+			return nil
+		}
+		return fmt.Errorf("Error reading the state of Batch account %q: %+v", name, err)
+	}
+
+	d.Set("name", resp.Name)
+	d.Set("resource_group_name", resourceGroupName)
+
+	if location := resp.Location; location != nil {
+		d.Set("location", azureRMNormalizeLocation(*location))
+	}
+
+	if props := resp.AccountProperties; props != nil {
+		if autoStorage := props.AutoStorage; autoStorage != nil {
+			d.Set("storage_account_id", autoStorage.StorageAccountID)
+		}
+		d.Set("pool_allocation_mode", props.PoolAllocationMode)
+	}
+
+	flattenAndSetTags(d, resp.Tags)
+
+	return nil
+}
+
+func resourceArmBatchAccountUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).batchAccountClient
+	ctx := meta.(*ArmClient).StopContext
+
+	log.Printf("[INFO] preparing arguments for Azure Batch account update.")
+
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	name := id.Path["batchAccounts"]
+	resourceGroupName := id.ResourceGroup
+
+	storageAccountId := d.Get("storage_account_id").(string)
+	tags := d.Get("tags").(map[string]interface{})
+
+	parameters := batch.AccountUpdateParameters{
+		AccountUpdateProperties: &batch.AccountUpdateProperties{
+			AutoStorage: &batch.AutoStorageBaseProperties{
+				StorageAccountID: &storageAccountId,
+			},
+		},
+		Tags: expandTags(tags),
+	}
+
+	_, err = client.Update(ctx, resourceGroupName, name, parameters)
+	if err != nil {
```

Comment:
_minor_ these liens could also be combined:
```
	if _, err = client.Update(ctx, resourceGroupName, name, parameters); err != nil {
```

Reply:
Done.